lib/WWW/Spotify.pm

```diff
@@ -1244,10 +1306,70 @@ needed for requests that require OAuth, see Spotify API documentation for more i
 
 Can also be set via environment variable, SPOTIFY_CLIENT_SECRET
 
+=head2 response_status
+
+returns the response code for the last request made
+
+    my $status = $spotify->response_status();
+
+=head2 response_content_type
+
+returns the response type for the last request made, helpful to verify JSON/XML
+
+    my $content_type = $spotify->response_content_type();
+
+=head2 request_custom_handler
+
+pass a callback subroutine to this method that will be run at the end of the
+request prior to check_response, if enabled
+
+    # $m is the WWW::Mechanize object
+    $spotify->request_custom_handler(
+        sub { my $m = shift;
+            if ($m->status() == 401) {
+                return 1;
+            }
+        }
+    );
+
+=head2 request_custom_handler_result
+
+returns the result of the most recent execution of the request_custom_handler callback
+this allows you to determine the success/failure criteria of your callback
+
+    my $callback_result = $spotify->request_custom_handler_result();
+
+=head2 check_response
```

Comment:
Given that it does on error, calling it `die_on_error` would make it obvious that some exception handling will need to be added by the implementor.